clustered/common/src/test/java/org/ehcache/clustered/common/ServerStoreCompatibilityTest.java

```diff
@@ -0,0 +1,485 @@
+  private static final PoolAllocation FIXED_POOL_ALLOCATION = new Fixed("primary",4);
+  private static final PoolAllocation SHARED_POOL_ALLOCATION = new Shared("sharedPool");
+  private static final String STORED_KEY_TYPE = Long.class.getName();
+  private static final String STORED_VALUE_TYPE = String.class.getName();
+  private static final String ACTUAL_KEY_TYPE = Long.class .getName();
+  private static final String ACTUAL_VALUE_TYPE = String.class.getName();
+  private static final String KEY_SERIALIZER_TYPE = Long.class.getName();
+  private static final String VALUE_SERIALIZER_TYPE = String.class.getName();
+
+  @Test
+  public void testStoredKeyTypeMismatch()
+  {
+    ServerStoreConfiguration serverConfiguration = new ServerStoreConfiguration(FIXED_POOL_ALLOCATION,
+                                                                                STORED_KEY_TYPE,
+                                                                                STORED_VALUE_TYPE,
+                                                                                ACTUAL_KEY_TYPE,
+                                                                                ACTUAL_VALUE_TYPE,
+                                                                                KEY_SERIALIZER_TYPE,
+                                                                                VALUE_SERIALIZER_TYPE,
+                                                                                Consistency.EVENTUAL);
+
+    ServerStoreConfiguration clientConfiguration = new ServerStoreConfiguration(FIXED_POOL_ALLOCATION,
+                                                                                String.class.getName(),
+                                                                                STORED_VALUE_TYPE,
+                                                                                ACTUAL_KEY_TYPE,
+                                                                                ACTUAL_VALUE_TYPE,
+                                                                                KEY_SERIALIZER_TYPE,
+                                                                                VALUE_SERIALIZER_TYPE,
+                                                                                Consistency.EVENTUAL);
+
+    ServerStoreCompatibility serverStoreCompatibility = new ServerStoreCompatibility();
+
+    try
+    {
+      serverStoreCompatibility.verify(serverConfiguration, clientConfiguration);
+      fail("Expected ClusteredStoreValidationException");
+    }
+    catch(ClusteredStoreValidationException e)
+    {
+      System.out.println("errorMsg: " + e.getMessage());
+      Assert.assertTrue("test failed", e.getMessage().equals(ERROR_MESSAGE_BASE + "storedKeyType existing: " + serverConfiguration.getStoredKeyType() + ", desired: " + clientConfiguration.getStoredKeyType()));
```

Comment:
I changed to check to use assertThat().  Why is this better?  Sorry this is my first time with junit.  More familiar with testng.